akka-http-tests/src/main/java/akka/http/javadsl/server/examples/simple/SimpleServerApp.java

```diff
@@ -4,84 +4,76 @@
 
-    static RequestVal<String> bodyAsName = RequestVals.entityAs(Unmarshallers.String());
+public class SimpleServerApp extends HttpApp {
 
-    public static RouteResult multiply(RequestContext ctx, int x, int y) {
+    public Route multiply(int x, int y) {
         int result = x * y;
-        return ctx.complete(String.format("%d * %d = %d", x, y, result));
+        return complete(String.format("%d * %d = %d", x, y, result));
     }
-    public static CompletionStage<RouteResult> multiplyAsync(final RequestContext ctx, final int x, final int y) {
-        return CompletableFuture.supplyAsync(() -> multiply(ctx, x, y), ctx.executionContext());
+    
+    public CompletionStage<Route> multiplyAsync(Executor ctx, int x, int y) {
+        return CompletableFuture.supplyAsync(() -> multiply(x, y), ctx);
     }
 
     @Override
     public Route createRoute() {
-        Handler addHandler = new Handler() {
-            @Override
-            public RouteResult apply(RequestContext ctx) {
-                int xVal = x.get(ctx);
-                int yVal = y.get(ctx);
-                int result = xVal + yVal;
-                return ctx.complete(String.format("%d + %d = %d", xVal, yVal, result));
-            }
+        Route addHandler = param(INT(), "x", x ->
+            param(INT(), "y", y -> {
+                int result = x + y;
+                return complete(String.format("%d + %d = %d", x, y, result));
+            })
+        );
+        
+        BiFunction<Integer, Integer, Route> subtractHandler = (x, y) -> {
+            int result = x - y;
+            return complete(String.format("%d - %d = %d", x, y, result));            
         };
-        Handler2<Integer, Integer> subtractHandler = new Handler2<Integer, Integer>() {
-            public RouteResult apply(RequestContext ctx, Integer xVal, Integer yVal) {
-                int result = xVal - yVal;
-                return ctx.complete(String.format("%d - %d = %d", xVal, yVal, result));
-            }
-        };
-        Handler1<String> helloPostHandler =
-            new Handler1<String>() {
-                @Override
-                public RouteResult apply(RequestContext ctx, String s) {
-                    return ctx.complete("Hello " + s + "!");
-                }
-            };
+        
         return
             route(
                 // matches the empty path
-                pathSingleSlash().route(
+                pathSingleSlash(() ->
                     getFromResource("web/calculator.html")
                 ),
                 // matches paths like this: /add?x=42&y=23
-                path("add").route(
-                    handleWith(addHandler, x, y)
-                ),
-                path("subtract").route(
-                    handleWith2(x, y, subtractHandler)
+                path("add", () -> addHandler),
+                path("subtract", () ->
+                    param(INT(), "x", x ->
+                        param(INT(), "y", y ->
```

Comment:
possible, but with `param2` you wouldn't even need the extra `named` calls, just `param2(INT, "x", INT, "y", (x,y) -> ...`. 

I wouldn't use it past 2 though: it pulls the argument order too far away from the place where they're declared. Should I add param2?